core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -458,7 +538,8 @@ private SubchannelData createNewSubchannel(SocketAddress addr, Attributes attrs)
   }
 
   private boolean isPassComplete() {
-    if (addressIndex.isValid() || subchannels.size() < addressIndex.size()) {
+    if ((!serializingRetries && addressIndex.isValid())
```

Comment:
Why do we want to ignore whether the index is valid?

Reply:
Because we reset it for subchannel retry.

Reply:
This is still nagging at me. I feel like either we never need this condition or we need to replace it with something else for serializingRetries. I agree that we need this function to return true even though we've reset the index (for `++numTf`).